app.py

```diff
@@ -1,13 +1,77 @@
-from flask import Flask
+from flask import Flask, jsonify
+import requests
+import os
 
 
 app = Flask(__name__)
 
+BASE_URL = os.environ.get('BASE_URL',
+                          'https://cplab1.techservices.illinois.edu:443/api')
+CLIENT_ID = os.environ.get('CLIENT_ID')
+CLIENT_SECRET = os.environ.get('CLIENT_SECRET')
+
 
-# Canary
 @app.route('/canary')
 def canary():
-    return 'Ok'
+    """
+    Check any external dependencies (api, database, etc...)
+    that the API has. If any dependency is unavailable return an
+    error message.
+    """
+    canary_result = 'Ok'
+    status_code = 200
+
+    # Check Clearpass API dependency
+    token_response = get_clearpass_token(CLIENT_ID, CLIENT_SECRET)
+
+    if not token_response['token'] or token_response['status_code'] != 200:
```

Comment:
I would get rid of this logic and add a try/catch around line 25. Or even better write a useable [decorator](https://realpython.com/primer-on-python-decorators/) to handle catching exceptions, and returns a 500 (or 503, etc.) or the functions behalf.